semantic/tools/semantic2vef.cpp

```diff
@@ -0,0 +1,247 @@
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "service/cmdline.hpp"
+
+#include <cstdlib>
+#include <iostream>
+
+#include <opencv2/core.hpp>
+#include <opencv2/imgcodecs.hpp>
+
+#include "utility/buildsys.hpp"
+#include "utility/gccversion.hpp"
+#include "utility/streams.hpp"
+#include "utility/path.hpp"
+#include "utility/implicit-value.hpp"
+#include "utility/small_map.hpp"
+
+#include "vef/vef.hpp"
+
+#include "service/cmdline.hpp"
+
+#include "geometry/meshop.hpp"
+
+#include "semantic/io.hpp"
+#include "semantic/mesh.hpp"
+
+namespace po = boost::program_options;
+namespace fs = boost::filesystem;
+
+namespace {
+
+const utility::small_map<semantic::Material, cv::Vec3b> materialColors(
+    { { semantic::Material::default_, cv::Vec3b(220, 220, 220) },
+      { semantic::Material::facade, cv::Vec3b(245, 245, 245) },
+      { semantic::Material::roof, cv::Vec3b(91, 114, 226) },
+      { semantic::Material::tree_trunk, cv::Vec3b(83, 53, 10) },
+      { semantic::Material::tree_crown_deciduous, cv::Vec3b(86, 137, 61) },
+      { semantic::Material::tree_crown_coniferous, cv::Vec3b(41, 73, 26) },
+      { semantic::Material::terrace, cv::Vec3b(71, 164, 233) } });
+
+class Semantic2Vef : public service::Cmdline
+{
+public:
+    Semantic2Vef()
+        : service::Cmdline("semantic2vef", BUILD_TARGET_VERSION)
+    {}
+
+private:
+    virtual void configuration(po::options_description &cmdline
+                               , po::options_description &config
+                               , po::positional_options_description &pd)
+        override;
+
+    virtual void configure(const po::variables_map &vars)
+        override;
+
+    virtual bool help(std::ostream &out, const std::string &what) const
+        override;
+
+    virtual int run() override;
+
+    semantic::MeshConfig meshConfig_;
+    std::vector<fs::path> input_;
+    fs::path output_;
+    int txtColorWidth_ = 8;
+};
+
+void Semantic2Vef::configuration(po::options_description &cmdline
+                        , po::options_description &config
+                        , po::positional_options_description &pd)
+{
+    cmdline.add_options()
+        ("input", po::value(&input_)->required()
+         , "Path to the input file with semantic data.")
+        ("output", po::value(&output_)->required()
+         , "Path to the output OBJ file.")
+        ("closedSurface"
+         , utility::implicit_value(&meshConfig_.closedSurface, true)
+         ->default_value(false)
+         , "Generate closed surface.")
+        ("txtColorWidth", po::value(&txtColorWidth_)
+         ->default_value(txtColorWidth_)
+         , "Width of individual color squares in generated texture image.")
+        ;
+
+    pd.add("output", 1).add("input", -1);
+
+    (void) config;
```

Comment:
I would prefer not to use not to give parameters we are not planning to use names. The `(void)` casting hack to silence the compiler warning is not the best practice.